pkg/proxy/ipvs/proxier.go

```diff
@@ -808,6 +808,42 @@ func (proxier *Proxier) syncProxyRules() {
 	// activeBindAddrs represents ip address successfully bind to DefaultDummyDevice in this round of sync
 	activeBindAddrs := map[string]bool{}
 
+	hasNodePort := false
+	for _, svc := range proxier.serviceMap {
+		svcInfo, ok := svc.(*serviceInfo)
+		if ok && svcInfo.NodePort() != 0 {
+			hasNodePort = true
+			break
+		}
+	}
+
+	// Both nodeAddresses and nodeIPs can be reused for all nodePort services
+	// and only need to be computed if we have at least one nodePort service.
+	var (
+		// List of node addresses to listen on if a nodePort is set.
+		nodeAddresses sets.String
+		// List of node IP addresses to be used as IPVS services if nodePort is
+		// set.
+		nodeIPs []net.IP
+	)
+
+	if hasNodePort {
+		nodeAddresses, err = utilproxy.GetNodeAddresses(proxier.nodePortAddresses, proxier.networkInterfacer)
+		if err != nil {
+			klog.Errorf("Failed to get node ip address matching nodeport cidr: %v", err)
+		}
+		for address := range nodeAddresses {
```

Comment:
use `nodeAddresses.List()` here

Reply:
Done, I actually changed `nodeAddresses` to a `[]string` that is only set if `nodeAddrSet.Len() > 0` to avoid multiple calls to `.List()` down the road.